docs/topics/request-response.rst

```diff
@@ -312,6 +315,40 @@ errors if needed::
                 request = failure.request
                 self.logger.error('TimeoutError on %s', request.url)
 
+.. _topics-request-response-ref-accessing-callback-arguments-in-errback:
+
+Accessing additional data in errback functions
+----------------------------------------------
+
+In case of a failure to process the request, you may be interested in
+accessing arguments to the callback functions so you can process further
+based on the arguments in the errback. The following example shows how to
+achieve this by using ``Failure.request.cb_kwargs``
+
+::
+
+    def parse(self, response):
+        request = scrapy.Request('http://www.example.com/index.html',
+                                 callback=self.parse_page2,
+                                 errback=self.errback_page2,
+                                 cb_kwargs=dict(main_url=response.url))
+        request.cb_kwargs['foo'] = 'bar'  # add more arguments for the callback
+        yield request
+
+    def parse_page2(self, response, main_url, foo):
+        yield dict(
+            main_url=main_url,
+            other_url=response.url,
+            foo=foo,
+        )
+
+    def errback_page2(self, failure):
+        yield dict(
+            main_url=failure.request.cb_kwargs['main_url'],
+            other_url=failure.value.response.url,
+            foo=failure.request.cb_kwargs['foo'],
+        )
```

Comment:
I think we should simplify this example further, remove parts that are irrelevant for this section.

Reply:
Thanks @Gallaecio for the comment, I think add "pass" to the parse_page2 function to simplify. Is there anything else you would you would like me to change to simplify?

``
def parse_page2(self, response, main_url, foo): pass
``

Reply:
I also noticed that `def parse`, for example, showcases how to define cb_kwargs 2 different ways. I don’t think that’s necessary here.

Also, I think a single argument should suffice for the demonstration. And that should allow making `errback_page2` shorter as well.

In general, looking at the example it should be a bit more obvious what the example is about, anything unrelated can distract user attention.

Reply:
@Gallaecio Done. Can you please review?